### PR TITLE
Update Harness.js

### PR DESCRIPTION
**Pull request by ali-uzma**
Broke down run function into separate functions

### CODE REVIEW

File: lib/Harness.js [hunk 1]
```diff
@@ -26,15 +26,28 @@ class Harness extends EventEmitter {
...
+  }
+  
+  _checkTestable() {
+    if (!this._testable || !this._testable._finalise) throw new Error('The harness must be initialised with a suite or test');
+  }
+  
+  _applyOptions(runtime) {
     const runtimeOptions = new Options({ runtime });
     const options = this._options.apply(runtimeOptions);
-
```
Comment (cressie176) [inline lib/Harness.js:37]: I think you meant to return the options, e.g.
```js
return this._options.apply(runtimeOptions);
```

**Review by cressie176 — CHANGES_REQUESTED** [review]
Thanks for the PR.  I think there's one small mistake, but otherwise looks good.

**Comment by ali-uzma** [comment]
Sorry about that, i forgot to return the options. I just updated the file. Hope it looks fine now.

**Comment by cressie176** [comment]
Thanks @Code-Science, I merged locally and have pushed to main. Not sure why GitHub didn't close the PR. I probably missed a command.

**Pull request closed without merging**